### PR TITLE
[ENT-491] Remove SSO consent features; verify fully-populated consent URL generation

### PR DESCRIPTION
**Pull request by haikuginger**
**Description:** This pull request removes the ability to request account-level consent via a linked SSO provider. Consent is now a separate operation.

**JIRA:** [ENT-491](https://openedx.atlassian.net/browse/ENT-491)

**Dependencies:** edx/edx-platform#15624

**Merge deadline:** 31 July 2017

**Installation instructions:** N/A

**Testing instructions:**

1. Verify that you can no longer set an EnterpriseCustomer to have "optional" or "at login" consent requirements.

**Merge checklist:**

- [ ] Check that the versions of the requirements in the `platform-master.in` file match edx-platform.
- [ ] New requirements are in the right place (`base.in` if only used in enterprise; in the correct `platform-****.in` files if they're hosted in edx-platform)
- [ ] Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are (reasonably) squashed
- [ ] Translations are updated
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)
- [ ] edx-platform PR (be sure to include edx-platform requirements upgrades that were present in this PR)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @haikuginger! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fedx-enterprise&number=152"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

File: enterprise/views.py [hunk 1]
```diff
@@ -360,81 +318,14 @@ def get_course_specific_consent(self, request, course_id):
...
-        Based on a URL parameter, the form rendered will either be course-specific or appropriate
-        for granting consent for all courses sponsored by the EnterpriseCustomer.
         """
         # Verify that all necessary resources are present
         verify_edx_resources()
         course = request.GET.get('course_id')
-        if course:
-            return self.get_course_specific_consent(request, course)
-        return self.get_account_consent(request)
+        return self.get_course_specific_consent(request, course)
```
Comment (brittneyexline) [inline enterprise/views.py]: what will happen if course is None? is it ever possible that it will be anymore?

File: enterprise/views.py [hunk 2]
```diff
@@ -550,10 +370,7 @@ def post(self, request):
         consent_provided = request.POST.get('data_sharing_consent', False)
         specific_course = request.POST.get('course_id')
 
-        if specific_course:
-            # We're handing consent only for a particular course
-            return self.post_course_specific_consent(request, specific_course, consent_provided)
-        return self.post_account_consent(request, consent_provided)
+        return self.post_course_specific_consent(request, specific_course, consent_provided)
```
Comment (brittneyexline) [inline enterprise/views.py]: same here

**Comment by haikuginger** [comment]
@brittneyexline, I updated the `.get()` call to default to an empty string; that'll cause it to fail when it goes to fetch course details. Sound good?

**Comment by haikuginger** [comment]
Rebased from e23b5ce.

**Pull request merged**